### PR TITLE
Changes to DSC Linux extension

### PR DESCRIPTION
**Pull request by balukambala**
1) Improved the tracing in DSC to debug/troubleshoot customer issues
2) Fixed few issues to exclude in score card where customer enters incorrect input.

### CODE REVIEW

File: DSC/dsc.py [hunk 1]
```diff
@@ -107,15 +107,19 @@ def get_distro_category():
...
 def install():
     hutil.do_parse_context('Install')
     try:
+        waagent.AddExtensionEvent(name=ExtensionShortName, op='InstallInprogress', isSuccess=True, message="Installing DSCForLinux extension")
         remove_old_dsc_packages()
         install_dsc_packages()
+        waagent.AddExtensionEvent(name=ExtensionShortName, op='InstallInprogress', isSuccess=True, message="successfully installed DSCForLinux extension")
         hutil.do_exit(0, 'Install', 'success', '0', 'Install Succeeded.')
     except Exception as e:
+        waagent.AddExtensionEvent(name=ExtensionShortName, op='InstallInprogress', isSuccess=True, message="failed to install an extension with error" + str(e))
```
Comment (boumenot) [inline DSC/dsc.py]: There are a few cases of this.

Installinprogress -> InstallInProgress

File: DSC/dsc.py [hunk 2]
```diff
@@ -125,16 +129,17 @@ def enable():
...
         else:
             mode = mode.lower()
             if not hasattr(Mode, mode):
                 waagent.AddExtensionEvent(name=ExtensionShortName,
                                           op=Operation.Enable,
-                                          isSuccess=False,
+                                          isSuccess=True,
                                           message="(03001)Argument error, invalid mode")
-                hutil.do_exit(1, 'Enable', 'error', '1', 'Enable failed, unknown mode: ' + mode)
+                hutil.do_exit(51, 'Enable', 'error', '51', 'Enable failed, unknown mode: ' + mode)
```
Comment (boumenot) [inline DSC/dsc.py]: Why is this 51?

File: DSC/dsc.py [hunk 3]
```diff
@@ -327,7 +340,8 @@ def get_openssl_version():
     else:
         error_msg = 'This system does not have a supported version of OpenSSL installed. Supported version: 0.9.8*, 1.0.*'
```
Comment (boumenot) [inline DSC/dsc.py]: 1.1.* is the latest version available, and should be supported.

File: DSC/dsc.py (hunk 2 above)
Comment (balukambala) [inline DSC/dsc.py]: DSC supports either pull or push mode. if users enters incorrect mode, then it is an argument error and exit code is 51. We don't want to consider this in our score card.

File: DSC/dsc.py (hunk 3 above)
Comment (balukambala) [inline DSC/dsc.py]: Currently, I am just improving the tracing and get hands on the release cycle. In the next iteration, I will test this if DSC bits can support 1.1.*

File: DSC/dsc.py [hunk 4]
```diff
@@ -174,6 +179,7 @@ def enable():
                 hutil.do_exit(1, 'Enable', 'error', '1', 'Enable failed. ' + current_config)
         hutil.do_exit(0, 'Enable', 'success', '0', 'Enable Succeeded')
     except Exception as e:
+        waagent.AddExtensionEvent(name=ExtensionShortName, op='EnableInprogress', isSuccess=True, message="Enable failed with the error " + str(e))
```
Comment (boumenot) [inline DSC/dsc.py]: EnableInprogress -> EnableInProgress

File: DSC/dsc.py (hunk 3 above)
Comment (boumenot) [inline DSC/dsc.py]: ack

**Pull request merged**